allure-notifications-api/src/main/java/guru/qa/allure/notifications/config/ApplicationConfig.java

```diff
@@ -1,36 +1,83 @@
 package guru.qa.allure.notifications.config;
 
-import java.io.FileNotFoundException;
-
 import guru.qa.allure.notifications.exceptions.ConfigNotFoundException;
 import guru.qa.allure.notifications.json.JSON;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.io.FileNotFoundException;
+import java.util.Properties;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
 
 /**
  * @author kadehar
  * @since 1.0
  * Utility class for config creation.
  */
 public class ApplicationConfig {
-    private static final ApplicationConfig INSTANCE = new ApplicationConfig();
     private static final String CONFIG_FILE_PROPERTY_NAME = "configFile";
-    private final String configFile = System.getProperty(CONFIG_FILE_PROPERTY_NAME);
 
-    private ApplicationConfig() {
+    private final String configFile;
+
+    public ApplicationConfig(String configFile) {
+        this.configFile = configFile;
     }
 
-    public static ApplicationConfig newInstance() {
-        return INSTANCE;
+    public ApplicationConfig() {
+        this(getConfigFile());
+    }
+
+    public static void apply(Config config, Properties properties) {
+        applySystemProperty(properties, "base.project", config.getBase()::setProject);
+        applySystemProperty(properties, "base.environment", config.getBase()::setEnvironment);
+        applySystemProperty(properties, "base.comment", config.getBase()::setComment);
+        applySystemProperty(properties, "base.reportLink", config.getBase()::setReportLink);
+        applySystemProperty(properties, "base.logo", config.getBase()::setLogo);
+        applySystemProperty(properties, "base.allureFolder", config.getBase()::setAllureFolder);
+        applySystemProperty(properties, "base.durationFormat", config.getBase()::setDurationFormat);
+
+        applySystemPropertyMap(properties, "base.customData.", config.getBase()::addCustomData);
+
+        applySystemProperty(properties, "telegram.token", config.getTelegram()::setToken);
+        applySystemProperty(properties, "telegram.chat", config.getTelegram()::setChat);
+        applySystemProperty(properties, "telegram.topic", config.getTelegram()::setTopic);
+        applySystemProperty(properties, "telegram.replyTo", config.getTelegram()::setReplyTo);
+        applySystemProperty(properties, "telegram.templatePath", config.getTelegram()::setTemplatePath);
```

Comment:
please use [Jackson JavaPropsMapper](https://www.javadoc.io/doc/com.fasterxml.jackson.dataformat/jackson-dataformat-properties/latest/com/fasterxml/jackson/dataformat/javaprop/JavaPropsMapper.html) to map the properties to the object (it will make the solution expandable)

Reply:
wow, works like a charm!